File: Archive-Code/LabPakStef.cpp
#include <iostream>
#include <cstring>
#include <string>
#include <map>

using namespace std;

class Encrypt{
    public:
    void make_dictionary(char start){
        string alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        int started = alphabet.find(start);
        char *huruf = new char[26];
        for (int l =0;l<26;l++){
            if (started == 26){
                started = started-26;
            }
            *huruf[l]=alphabet[started];
            started += 1;
        }
        dictionary[start]=huruf;
        delete huruf;
        cout << dictionary.size() << "\n";
        for (map<char,char*>::iterator k=dictionary.begin();k!=dictionary.end(); ++k){
            cout << k->first << " " << k->second << "\n";
        }
        
    }
    private:
    map<char,char*> dictionary;
};

int main(){
    map<char,string> dict_key;
    string key,text;
    Encrypt enkripsi;
    cin >> key;
    for (string::iterator it=key.begin();it!=key.end();++it){
        cout << *it;
        enkripsi.make_dictionary(*it);
        cout << "\n";   
        }
}

File: Archive-Code/LabPakStef.py
import string
class Encrypt:    
    def make_dictionary(self,start):
        alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        self.dictionary = {}
        started = alphabet.index(start)
        for i in alphabet:
            if started == 26  :
                started = started-26
            self.dictionary[i]=alphabet[started]
            started += 1

    def run(self,huruf):
        return self.dictionary[huruf]

class Decrypt:
    def make_dictionary(self,start):
        alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        self.dictionary = {}
        started = alphabet.index(start)
        for i in alphabet:
            if started == 26  :
                started = started-26
            self.dictionary[alphabet[started]]=i
            started += 1
    
    def run(self,huruf):
        return self.dictionary[huruf]


dict_key = {}
key = input("Masukkan key : ").upper()
text = input("Masukkan text : ").upper()
perintah = input("D(Decrypt) or E(Encrypt)? ")
if perintah.upper() == "E":
    for k in key:
            dict_key[k]= Encrypt()
            dict_key[k].make_dictionary(k)

    for k in range(len(text)):
        if text[k] == " " or text[k] in string.punctuation:
            print(text[k],end="")
        else :
            temp = k % len(key)
            print(dict_key[key[temp]].run(text[k]),end="")
elif perintah.upper() == "D" :
    for k in key:
            dict_key[k]= Decrypt()
            dict_key[k].make_dictionary(k)
    for k in range(len(text)):
        if text[k] == " " or text[k] in string.punctuation:
            print(text[k],end="")
        else :
            temp = k % len(key)
            print(dict_key[key[temp]].run(text[k]),end="")
else :
    print('Tidak ada Perintah')
